### ISSUE TITLE
NaN leak with selectdim view

### ISSUE THREAD

**Issue by michaelsachs** [brenhinkeller/NaNStatistics.jl#39]
I came across an example where NaNs "leak" into the result of a `nanmean` operation. This occurs when I average on a view created using `selectdim`.

```
# create a matrix and set one element to NaN
a = rand(10,5)
a[3,1] = NaN
```

When I create a view using the appropriate macro everything looks fine:
```
av = @view a[:,1:3]
avMean = nanmean(av,dims=2)
avMean[3,1] 
0.8755526000081276
```

However, when I create the same view using `selectdim` the NaN ends up in the final vector:
```
av = selectdim(a,2,[true,true,true,false,false])
avMean = nanmean(av,dims=2)
avMean[3,1]
NaN
```

I am using Julia 1.9.0 and  NaNStatistics v0.6.28.

**Comment by brenhinkeller** [comment]
Thanks for the report! I think the immediate solution will be to just be stricter about what non-`Array` array types we use `@turbo` on

**Comment by brenhinkeller** [comment]
Should be fixed in `v0.6.29` and onwards
